src/libfetchers/git.cc

```diff
@@ -432,6 +432,29 @@ struct GitInputScheme : InputScheme
             return {std::move(storePath), input};
         };
 
+        if (auto narHash = input.getNarHash()) {
+            auto expectedPath = store->makeFixedOutputPath(name, FixedOutputInfo {
+                .hash = {
+                    .method = FileIngestionMethod::Recursive,
+                    .hash = *narHash,
+                },
+                .references = {},
+            });
+
+            try {
+                store->ensurePath(expectedPath);
+
+                Attrs attrs({
+                    // TODO: do these need real values?
+                    {"lastModified", 0ull},
+                    {"revCount", 0ull},
```

Comment:
We definitely cannot provide fake values depending on whether the path was substituted. That would make evaluation non-deterministic.

Reply:
`lastModified` and `revCount` are both calculated from the git repo, which we don't have access to in the case of substitution (just the tree checkout, not the .git dir). Are those values written into a .drv file in the binary cache (I'm new to nix)?

If not, then I think we could require that the user specify narHash+lastModified+revCount if they want substitution to work.

Reply:
You could make these values lazy, and retrieve them as usual, but only when needed. Though laziness isn't supported by libfetchers yet(?). `nix::fetchers::Attrs` currently only allows data; no callbacks.